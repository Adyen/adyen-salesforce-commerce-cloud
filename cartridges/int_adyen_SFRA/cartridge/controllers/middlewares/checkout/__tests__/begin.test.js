"use strict";

/* eslint-disable global-require */
var _require = require('../../index'),
  begin = _require.checkout.begin;
var OrderMgr = require('dw/order/OrderMgr');
var BasketMgr = require('dw/order/BasketMgr');
var ShippingMgr = require('dw/order/ShippingMgr');
var Logger = require('dw/system/Logger');
var Transaction = require('dw/system/Transaction');
var res;
var req;
beforeEach(function () {
  jest.clearAllMocks();
  req = {
    currentCustomer: {
      raw: {
        isAuthenticated: jest.fn(function () {
          return false;
        })
      }
    },
    session: {
      privacyCache: {
        get: jest.fn(),
        set: jest.fn()
      }
    }
  };
  res = {
    getViewData: jest.fn(function () {
      return {};
    }),
    setViewData: jest.fn()
  };
});
describe('Begin', function () {
  it('should update saved cards', function () {
    var _require2 = require('*/cartridge/scripts/updateSavedCards'),
      updateSavedCards = _require2.updateSavedCards;
    req.currentCustomer.raw.isAuthenticated.mockImplementation(function () {
      return true;
    });
    begin(req, res, jest.fn());
    expect(updateSavedCards).toBeCalledTimes(1);
  });
  it('should set view data', function () {
    begin(req, res, jest.fn());
    expect(res.setViewData.mock.calls).toMatchSnapshot();
  });
  it('should not attempt to restore cart when no order number is cached', function () {
    begin(req, res, jest.fn());
    expect(res.setViewData.mock.calls).toMatchSnapshot();
    expect(OrderMgr.failOrder).not.toHaveBeenCalled();
  });
  it('should not attempt to restore cart when cart is not empty', function () {
    req.session.privacyCache.get.mockImplementationOnce(function () {
      return '12312';
    });
    OrderMgr.status = {
      value: "0"
    };
    begin(req, res, jest.fn());
    expect(Transaction.wrap).not.toHaveBeenCalled();
  });
  it('should successfully restore cart when current cart is empty and order number is in cache', function () {
    req.session.privacyCache.get.mockImplementationOnce(function () {
      return "0";
    });
    BasketMgr.getAllProductLineItems.mockImplementationOnce(function () {
      return [];
    });
    begin(req, res, jest.fn());
    expect(Transaction.wrap).toHaveBeenCalled();
  });
});