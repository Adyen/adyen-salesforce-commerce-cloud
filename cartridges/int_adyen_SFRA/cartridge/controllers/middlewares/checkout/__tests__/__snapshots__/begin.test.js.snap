// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Begin should not attempt to restore cart when no order number is cached 1`] = `
[
  [
    {
      "adyen": {
        "SFRA6Enabled": false,
        "adyenClientKey": "mocked_client_key",
        "cardholderNameBool": true,
        "clientKey": "mocked_client_key",
        "environment": "test",
        "googleMerchantID": "mocked_google_merchant_id",
        "installments": true,
        "merchantAccount": "mocked_merchant_account",
      },
    },
  ],
]
`;

exports[`Begin should set view data 1`] = `
[
  [
    {
      "adyen": {
        "SFRA6Enabled": false,
        "adyenClientKey": "mocked_client_key",
        "cardholderNameBool": true,
        "clientKey": "mocked_client_key",
        "environment": "test",
        "googleMerchantID": "mocked_google_merchant_id",
        "installments": true,
        "merchantAccount": "mocked_merchant_account",
      },
    },
  ],
]
`;
